### PR TITLE
A note about optional application components

### PR DESCRIPTION
**Pull request by divolgin**

### CODE REVIEW

File: docs/enterprise/image-registry-embedded-cluster.md [hunk 1]
```diff
@@ -42,6 +42,11 @@ Registry username and password can be found in the `registry-creds` secret in th
 ### Limitations
 Currently the image garbage collection feature has following limitations:
 
+#### Optional Components
+Some applications define Kubernetes resources that can be enabled or disabled dynamically. For example, template functions can be used to conditionally deploy a StatefulSet based on configuration from the user.
+If a resource is disabled and no longer deployed, its images may be garbage collected.
+To prevent this from happening, optional images should be included in the `additionalImages`](/reference/custom-resource-application#additionalimages) list.
```
Comment (ajp-io) [inline docs/enterprise/image-registry-embedded-cluster.md:48]: @replicatedhq/replicated-docs I know this isn't how you want to do links, but I'm going to leave it to you to update it to the format you want.

**Review by ajp-io — APPROVED** [review]

**Review by ajp-io — APPROVED** [review]

**Review by jonquil2002 — DISMISSED** [review]
Committed some wording/style changes, but otherwise LGTM.

**Comment by ajp-io** [comment]
@jonquil2002 "Garbage collected" is standard language for this. For example, you can see in [this section](https://en.wikipedia.org/wiki/Garbage_collection_(computer_science)#Tracing) of the Wikipedia page they say "garbage collected." I think the original wording would be more accurate and understandable to our audience than "added to the garbage collection."

**Review by jonquil2002 — APPROVED** [review]
@ajp-io  Reverted the wording as you suggested but kept the cross-reference link formatting.

**Comment by ajp-io** [comment]
Thanks, JQ :)

**Pull request merged**